tests/integration/devfile/cmd_devfile_delete_test.go

```diff
@@ -31,187 +38,250 @@ var _ = Describe("odo devfile delete command tests", func() {
-			helper.CmdShouldPass("odo", "create", "nodejs", "--project", commonVar.Project, componentName)
+	When("a component is created", func() {
+		JustBeforeEach(func() {
+			helper.CmdShouldPass("odo", "create", "nodejs", componentName, "--project", commonVar.Project, "--app", appName)
 			helper.CopyExample(filepath.Join("source", "devfiles", "nodejs", "project"), commonVar.Context)
-			helper.CopyExampleDevFile(filepath.Join("source", "devfiles", "nodejs", "devfile.yaml"), filepath.Join(commonVar.Context, "devfile.yaml"))
-			helper.CmdShouldPass("odo", "delete", "--project", commonVar.Project, "-f")
 		})
-
-		It("should delete the component created from the devfile and also the owned resources", func() {
-			resourceTypes := []string{"deployments", "pods", "services", "ingress"}
-
-			helper.CmdShouldPass("odo", "create", "nodejs", "--project", commonVar.Project, componentName)
-			helper.CopyExample(filepath.Join("source", "devfiles", "nodejs", "project"), commonVar.Context)
-			helper.CopyExampleDevFile(filepath.Join("source", "devfiles", "nodejs", "devfile.yaml"), filepath.Join(commonVar.Context, "devfile.yaml"))
-			helper.CmdShouldPass("odo", "url", "create", "example", "--host", "1.2.3.4.nip.io", "--port", "3000", "--ingress")
-
-			if os.Getenv("KUBERNETES") != "true" {
-				helper.CmdShouldPass("odo", "url", "create", "example-1", "--port", "3000")
-				resourceTypes = append(resourceTypes, "routes")
-			}
-
-			helper.CmdShouldPass("odo", "push", "--project", commonVar.Project)
-
-			helper.CmdShouldPass("odo", "delete", "--project", commonVar.Project, "-f")
-
-			for _, resourceType := range resourceTypes {
-				commonVar.CliRunner.WaitAndCheckForExistence(resourceType, commonVar.Project, 1)
-			}
+		JustAfterEach(func() {})
+		It("should delete the component", func() {
+			helper.CmdShouldPass("odo", "delete", "-f")
 		})
 
-		It("should delete the component created from the devfile and also the env and odo folders and the odo-index-file.json file with all flag", func() {
-			helper.CmdShouldPass("odo", "create", "nodejs", "--project", commonVar.Project, componentName)
-			helper.CopyExample(filepath.Join("source", "devfiles", "nodejs", "project"), commonVar.Context)
-			helper.CopyExampleDevFile(filepath.Join("source", "devfiles", "nodejs", "devfile.yaml"), filepath.Join(commonVar.Context, "devfile.yaml"))
-			helper.CmdShouldPass("odo", "push", "--project", commonVar.Project)
-
-			helper.CmdShouldPass("odo", "url", "create", "example", "--host", "1.2.3.4.nip.io", "--port", "3000", "--ingress")
-
-			if os.Getenv("KUBERNETES") != "true" {
-				helper.CmdShouldPass("odo", "url", "create", "example-1", "--port", "3000")
-			}
+		When("the component is pushed", func() {
+			JustBeforeEach(func() {
+				helper.CmdShouldPass("odo", "push", "--project", commonVar.Project)
+			})
+			JustAfterEach(func() {})
+			It("should delete the component, env, odo folders and odo-index-file.json with --all flag", func() {
```

Comment:
Can this test be placed before to run `odo push`?

If `odo push` is not necessary, we can save a couple of minutes ;)

Reply:
I had the same concern/thought as you, but `odo push` creates the odo-index-file.json which we require for this test, so we will need to execute it. :)